frontend/src/rauta.tsx

```diff
@@ -93,8 +93,29 @@ export type RouterLib = {
     Router: (props: RouterProps) => JSX.Element;
 };
 
-export type Listener = () => void;
-export type BeforeNavigationListener = (preventNavigation: () => void) => void;
+/** Helper class: a list of listeners */
+class Listeners<F extends (...args: any) => any> {
+    public list: { listener: F }[] = [];
+
+    /** Adds a new listener. Returns function to remove that listener again. */
+    public add(listener: F): () => void {
+        const obj = { listener };
```

Comment:
Why this wrapping?

Reply:
I added a comment in my last commit which should explain this